ably/auth.go

```diff
@@ -360,9 +352,9 @@ func (a *Auth) setDefaults(opts *authOptions, req *TokenRequest) error {
 }
 
 //Timestamp returns the timestamp to be used in authorization request.
-func (a *Auth) timestamp(query bool) (time.Time, error) {
+func (a *Auth) timestamp(queryCtx context.Context) (time.Time, error) {
 	now := a.client.opts.Now()
-	if !query {
+	if queryCtx == nil {
```

Comment:
I'm not sure I agree with replacing a boolean with a nil check on the context (since it's bad practice to pass a `nil` context), why not continue to use the boolean argument?

Reply:
I was on the fence for this one because an extra boolean flag that can just be a nil check on the context is redundant. But you're right it feels weird. Done at cf66a78.

Reply:
👍